kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -429,7 +429,7 @@ public static Optional<Tuple2<Protocol, Set<TableFeature>>> autoUpgradeProtocolB
 
     Set<TableFeature> allNeededTableFeatures =
         extractAllNeededTableFeatures(newMetadata, currentProtocol);
-    if (needDomainMetadataSupport && !allNeededTableFeatures.contains(DOMAIN_METADATA_W_FEATURE)) {
+    if (needDomainMetadataSupport) {
       allNeededTableFeatures =
```

Comment:
not blocking but why not just `allNeededTableFreatures.add(domainMetadata)`?

Reply:
It may not be mutable Set given we are using Streams